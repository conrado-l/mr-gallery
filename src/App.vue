<template>
  <v-app>
    <Navigation></Navigation>

    <v-main>
      <router-view v-if="getIsUserAuthenticated"></router-view>
      <div v-else-if="!errorAuthenticating" class="mt-5">
        <div class="d-flex justify-center align-center">
          <v-progress-circular
            :size="50"
            color="secondary"
            indeterminate
          ></v-progress-circular>
          <span class="ml-3"> we're getting ready for you! please wait </span>
        </div>
      </div>
    </v-main>
  </v-app>
</template>

<script>

import { mapGetters } from 'vuex'
import Navigation from '@/components/Navigation'

/** The app's main component **/
export default {
  name: 'App',
  components: { Navigation },
  data: () => ({
    errorAuthenticating: false,
    showSnackbar: false
  }),
  computed: {
    ...mapGetters('auth', [
      'getIsUserAuthenticated'
    ])
  },
  /** Created life-cycle hook */
  created () {
    this.$store.dispatch('auth/fetchToken')
      .then(() => {
        this.$toast('Welcome back!', { x: 'center', color: 'secondary' })
      })
      .catch(() => {
        this.$toast.error('An error has occurred while trying to authenticate')
        this.errorAuthenticating = true
      })
  }
}
</script>
