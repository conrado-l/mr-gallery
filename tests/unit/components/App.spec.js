// Vue Test Utils
import { createLocalVue, mount } from '@vue/test-utils'

// Vuex
import { Store } from 'vuex-mock-store'

// Plugins
import Vuetify from 'vuetify'

// Components
import App from '@/App'

describe('App.vue', () => {
  const localVue = createLocalVue()
  let vuetify

  let store

  // Reusable factory mount
  const factoryMount = () => {
    return mount(App, {
      localVue,
      vuetify,
      stubs: ['router-view'],
      mocks: {
        $store: store,
        $toast: {
          error: jest.fn()
        }
      }
    })
  }

  beforeEach(() => {
    vuetify = new Vuetify()

    store = new Store({
      state: {
        auth: {
          token: null,
          authenticated: false,
          fetching: false
        }
      },
      getters: {
        'auth/getToken': null,
        'auth/getIsUserAuthenticated': false,
        'auth/getFetchingState': false
      }
    })

    store.dispatch = jest.fn().mockResolvedValue('data') // dispatch has to return a promise
  })

  afterEach(() => {
    store.reset()
  })

  it('should render the component correctly', () => {
    // Instantiate the component
    const wrapper = factoryMount()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should initially call the fetch token action', () => {
    // Instantiate the component
    factoryMount()

    // Check if the action was dispatched
    expect(store.dispatch).toHaveBeenCalledWith('auth/fetchToken')
  })
})
