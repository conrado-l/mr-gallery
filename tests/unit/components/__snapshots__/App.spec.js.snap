// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should render the component correctly 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <header
      class="d-flex justify-center v-sheet theme--dark v-toolbar v-app-bar v-app-bar--fixed primary"
      style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;"
    >
      <div
        class="v-toolbar__content"
        style="height: 64px;"
      >
        <div
          class="v-toolbar__title"
        >
          Madison Reed | Gallery
        </div>
      </div>
    </header>
     
    <main
      class="v-main"
      style="padding: 64px 0px 0px 0px;"
    >
      <div
        class="v-main__wrap"
      >
        <div
          class="mt-5"
        >
          <div
            class="d-flex justify-center align-center"
          >
            <div
              aria-valuemax="100"
              aria-valuemin="0"
              class="v-progress-circular v-progress-circular--indeterminate secondary--text"
              role="progressbar"
              style="height: 50px; width: 50px;"
            >
              <svg
                style="transform: rotate(0deg);"
                viewBox="21.73913043478261 21.73913043478261 43.47826086956522 43.47826086956522"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  class="v-progress-circular__overlay"
                  cx="43.47826086956522"
                  cy="43.47826086956522"
                  fill="transparent"
                  r="20"
                  stroke-dasharray="125.664"
                  stroke-dashoffset="125.66370614359172px"
                  stroke-width="3.4782608695652177"
                />
              </svg>
              <div
                class="v-progress-circular__info"
              />
            </div>
             
            <span
              class="ml-3"
            >
               we're getting ready for you! please wait 
            </span>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
